pkg/git/k8s_git_keychain.go

```diff
@@ -13,22 +12,18 @@ import (
 	"github.com/pivotal/kpack/pkg/secret"
 )
 
-type k8sGitKeychain struct {
+type k8sGitKeychainFactory struct {
 	secretFetcher secret.Fetcher
 }
 
-var anonymousAuth transport.AuthMethod = nil
-
-func newK8sGitKeychain(k8sClient k8sclient.Interface) *k8sGitKeychain {
-	return &k8sGitKeychain{secretFetcher: secret.Fetcher{Client: k8sClient}}
+func newK8sGitKeychainFactory(k8sClient k8sclient.Interface) *k8sGitKeychainFactory {
+	return &k8sGitKeychainFactory{secretFetcher: secret.Fetcher{Client: k8sClient}}
 }
 
-func (k *k8sGitKeychain) Resolve(ctx context.Context, namespace, serviceAccount string, git v1alpha1.Git) (transport.AuthMethod, error) {
+func (k *k8sGitKeychainFactory) KeychainForServiceAccount(ctx context.Context, namespace, serviceAccount string) (GitKeychain, error) {
 	secrets, err := k.secretFetcher.SecretsForServiceAccount(ctx, serviceAccount, namespace)
 	if err != nil && !k8serrors.IsNotFound(err) {
```

Comment:
do we care about `IsNotFound` error handling?

Reply:
I believe the idea is a not found error will result in an empty git keychain now.

Reply:
Previously, this method was the resolve method that would return the actual auth. Now it just returns the keychain so we are delaying the error until libgit2 asks for auth and we can't provide it.

Reply:
Perhaps we can clarify that though. Let me ponder it.